crates/aptos-inspection-service/src/server/peer_information.rs

```diff
@@ -31,86 +38,263 @@ pub fn handle_peer_information_request(
+        &mut peer_information_output,
+        registered_networks,
+        &peers_and_metadata,
+    );
+    peer_information_output.push("\n".into());
+
+    // Display basic peer metadata for each peer
+    display_peer_monitoring_metadata(
+        &mut peer_information_output,
+        &all_peers,
+        &peers_and_metadata,
+    );
+    peer_information_output.push("\n".into());
+
+    // Display state sync metadata for each peer
+    display_state_sync_metadata(&mut peer_information_output, &all_peers, aptos_data_client);
+    peer_information_output.push("\n".into());
+
+    // Display detailed peer metadata for each peer
+    display_detailed_monitoring_metadata(
+        &mut peer_information_output,
+        &all_peers,
+        &peers_and_metadata,
+    );
+    peer_information_output.push("\n".into());
+
+    // Display the internal client state for each peer
+    display_internal_client_state(
+        &mut peer_information_output,
+        &all_peers,
+        &peers_and_metadata,
+    );
+
+    peer_information_output.join("\n") // Separate each entry with a newline to construct the output
+}
+
+/// Displays detailed peer monitoring metadata for each peer
+fn display_detailed_monitoring_metadata(
+    peer_information_output: &mut Vec<String>,
+    all_peers: &Vec<PeerNetworkId>,
+    peers_and_metadata: &Arc<PeersAndMetadata>,
+) {
+    peer_information_output.push("Detailed monitoring metadata for each peer:".into());
+
+    // Fetch and display the detailed metadata for each peer
+    for peer in all_peers {
+        if let Ok(peer_metadata) = peers_and_metadata.get_metadata_for_peer(*peer) {
+            let peer_monitoring_metadata = peer_metadata.get_peer_monitoring_metadata();
+            peer_information_output.push(format!(
+                "\t- Peer: {}, detailed metadata: {:?}", // Debug formatting for detailed metadata
+                peer, peer_monitoring_metadata
+            ));
+        }
+    }
+}
+
+/// Displays the internal client state for each peer
+fn display_internal_client_state(
+    peer_information_output: &mut Vec<String>,
+    all_peers: &Vec<PeerNetworkId>,
+    peers_and_metadata: &Arc<PeersAndMetadata>,
+) {
+    peer_information_output.push("Internal client state for each peer:".into());
+
+    // Fetch and display the internal client state for each peer
+    for peer in all_peers {
+        if let Ok(peer_metadata) = peers_and_metadata.get_metadata_for_peer(*peer) {
+            let peer_monitoring_metadata = peer_metadata.get_peer_monitoring_metadata();
+            peer_information_output.push(format!(
+                "\t- Peer: {}, internal client state: {:?}",
+                peer, peer_monitoring_metadata.internal_client_state
+            ));
+        }
+    }
+}
+
+/// Displays connection metadata for each peer
+fn display_peer_connection_metadata(
+    peer_information_output: &mut Vec<String>,
+    all_peers: &Vec<PeerNetworkId>,
+    peers_and_metadata: &Arc<PeersAndMetadata>,
```

Comment:
nit: &Arc is weird, you probably want to take &PeersAndMetadata here and other display functions.

Reply:
I'm not sure it really matters 😄 But, I've changed it to `&PeersAndMetadata`